includes/ExternalRedirectInterceptor.php

```diff
@@ -0,0 +1,65 @@
+<?php
+namespace NewfoldLabs\WP\Module\Onboarding;
+
+use NewfoldLabs\WP\Module\Onboarding\Data\Data;
+
+/**
+ * Class to intercept redirect calls and filter them.
+ *
+ * The purpose of this class is to prevent any redirects while the user is on the onboarding page.
+ * The only allowed redirect is to the brand plugin page.
+ */
+class ExternalRedirectInterceptor {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		if ( ! isset( $_GET['page'] ) || \sanitize_text_field( $_GET['page'] ) !== WP_Admin::$slug ) {
+			return;
+		}
+
+		\add_filter( 'wp_redirect', array( $this, 'wp_redirect' ), 10, 1 );
+	}
+
+	/**
+	 * Intercept wp_redirect calls and filter them.
+	 *
+	 * @param string $location The location to redirect to.
+	 */
+	public function wp_redirect( $location ): string {
+		$runtime_data     = Data::runtime();
+		$brand_plugin_url = '';
+
+		/*
+		 * Get the brand plugin page URL from the runtime data.
+		 */
+		// Check if the current brand is set and is an array.
+		if (
+			isset( $runtime_data['currentBrand'] ) &&
+			is_array( $runtime_data['currentBrand'] )
+			) {
+			// Check if the pluginDashboardPage key is set and is a string.
+			if (
+				isset( $runtime_data['currentBrand']['pluginDashboardPage'] ) &&
+				is_string( $runtime_data['currentBrand']['pluginDashboardPage'] )
```

Comment:
```suggestion
			isset( $runtime_data['currentBrand'], $runtime_data['currentBrand']['pluginDashboardPage'] ) &&
			is_string( $runtime_data['currentBrand']['pluginDashboardPage'] )
```
@wpalani  We can probably simplify to this